src/main/java/core/basesyntax/service/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service.strategy.impl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        if (transaction == null) {
+            throw new NullPointerException("Transaction cannot be null");
```

Comment:
Don't throw NullPointerException. It's a system exception. Better use IllegalArgumentException or create a custom exception